src/iotEdgeExplorer.ts

```diff
@@ -116,6 +116,39 @@ export class IoTEdgeExplorer extends BaseExplorer {
         return true;
     }
 
+    private isValidCreateOptions(desiredProperties: any): boolean {
+        const systemModules = desiredProperties.systemModules;

```

Comment:
Too many duplicated code, put wrap into a function for the similar code. Or `Cannot convert createOptions of custom module "${moduleName}" as json object: ${error.message}` is enough.

Reply:
Add a helper function